syncd/syncd_notifications.cpp

```diff
@@ -66,199 +66,212 @@ sai_fdb_entry_type_t getFdbEntryType(
-            for (const auto &fdbkey: g_redisClient->keys(pattern))
+            /* we only remove dynamic fdb entries here, static fdb entries need to be deleted manually by user instead of flush */
+            auto pEntryType = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_TYPE");
+
+            if (pEntryType && *pEntryType == strEntryType)
             {
-                /* we only remove dynamic fdb entries here, static fdb entries need to be deleted manually by user instead of flush */
-                auto pEntryType = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_TYPE");
+                auto bp = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID");
 
-                if (pEntryType && *pEntryType == strEntryType)
+                if (bp && *bp == strBridgePortId)
                 {
                     SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_FLUSHED",fdbkey.c_str());
                     g_redisClient->del(fdbkey);
                 }
             }
         }
-        else
-        {
-            SWSS_LOG_NOTICE("received a flush fdb event, port_oid = 0x%" PRIx64 ", bv_id = 0x%" PRIx64, port_oid, bv_id);
+    }
+}
 
-            auto strBridgeVlanId = sai_serialize_object_id(bv_id);
-            auto strBridgePortId = sai_serialize_object_id(port_oid);
 
-            std::string pattern = ASIC_STATE_TABLE + std::string(":SAI_OBJECT_TYPE_FDB_ENTRY:*") + strBridgeVlanId + "*";
-            for (const auto &fdbkey: g_redisClient->keys(pattern))
-            {
-                /* we only remove dynamic fdb entries here, static fdb entries need to be deleted manually by user instead of flush */
-                auto pEntryType = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_TYPE");
+void redisPutFdbEntryToAsicView(
+        _In_ const sai_fdb_event_notification_data_t *fdb)
+{
+    SWSS_LOG_ENTER();
 
-                if (pEntryType && *pEntryType == strEntryType)
-                {
-                    auto bp = g_redisClient->hget(fdbkey, "SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID");
+    // NOTE: this fdb entry already contains translated RID to VID
 
-                    if (bp && *bp == strBridgePortId)
-                    {
-                        SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_FLUSHED",fdbkey.c_str());
-                        g_redisClient->del(fdbkey);
-                    }
-                }
-            }
-        }
+    std::vector<swss::FieldValueTuple> entry;
+
+    entry = SaiAttributeList::serialize_attr_list(
+            SAI_OBJECT_TYPE_FDB_ENTRY,
+            fdb->attr_count,
+            fdb->attr,
+            false);
+
+    sai_object_type_t objectType = SAI_OBJECT_TYPE_FDB_ENTRY;
+
+    std::string strObjectType = sai_serialize_object_type(objectType);
+
+    std::string strFdbEntry = sai_serialize_fdb_entry(fdb->fdb_entry);
+
+    std::string key = ASIC_STATE_TABLE + (":" + strObjectType + ":" + strFdbEntry);
```

Comment:
>ASIC_STATE_TABLE [](start = 22, length = 16)

Seem you add `()` to convert `char *` into `string`. More readable way is explicitly convert first one into string.

Reply:
not the scope of this PR, i extracted part of this function to separate method, but somehow git diff messed things around since those 2 functions are too close to each other in code